x-pack/qa/full-cluster-restart/src/test/java/org/elasticsearch/xpack/restart/MLModelDeploymentFullClusterRestartIT.java

```diff
@@ -92,6 +91,10 @@ public void testDeploymentSurvivesRestart() throws Exception {
             startDeployment(modelId);
             assertInfer(modelId);
         } else {
+            ensureHealth(".ml-inference-*,.ml-config*", (request -> {
```

Comment:
This should not be required with the other changes.

Reply:
@davidkyle It ensures that the index is around before even bothering to wait around for the model. Correct, it doesn't do anything for the node assignment/allocation. But since the trigger trained model loading is in 5s intervals, I thought it good to not bother waiting for that until shards are available.